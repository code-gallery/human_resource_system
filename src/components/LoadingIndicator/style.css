.LoadingIndicator {
  margin: 0 0 2em;
  height: 100px;
  width: 20%;
  text-align: center;
  padding: 1em;
  margin: 0 auto 1em;
  vertical-align: top;
}

.LoadingIndicator-block {
  display: block;
  padding-top: 130px;
}

.LoadingIndicator-inline-block {
  display: inline-block;
}

.LoadingIndicator svg {
  enable-background: 0 0 50 50;
}

.LoadingIndicator svg path,
.LoadingIndicator svg rect {
  fill: #FF6700;
}
