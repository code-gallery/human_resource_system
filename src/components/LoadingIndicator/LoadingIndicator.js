import React from 'react'
import PropTypes from 'prop-types'
import Loader from 'components/Loader'
import './style.css'

const LoadingIndicator = (props) => {
  const { size, display } = props

  return (
    <div className={`LoadingIndicator LoadingIndicator-${display}`}>
      <Loader size={size} color="#FF6700" />
    </div>
  )
}

LoadingIndicator.defaultProps = {
  size: '40',
  display: 'block'
}

LoadingIndicator.propTypes = {
  size: PropTypes.string,
  display: PropTypes.string
}

export default LoadingIndicator
