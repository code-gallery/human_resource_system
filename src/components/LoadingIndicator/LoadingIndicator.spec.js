import React from 'react'
import renderer from 'react-test-renderer'
import LoadingIndicator from './index'

describe('<LoadingIndicator />', () => {
  describe('without any props', () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <LoadingIndicator />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })

  describe('with props', () => {
    it('renders correctly', () => {
      const props = {
        size: '80',
        display: 'inline'
      }

      const tree = renderer.create(
        <LoadingIndicator {...props} />
      ).toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
