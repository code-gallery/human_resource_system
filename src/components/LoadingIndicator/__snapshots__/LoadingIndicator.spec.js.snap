// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> with props renders correctly 1`] = `
<div
  className="LoadingIndicator LoadingIndicator-inline"
>
  <svg
    className=""
    height="80px"
    version="1.1"
    viewBox="0 0 50 50"
    width="80px"
    x="0px"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    y="0px"
  >
    <path
      d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
      fill="#FF6700"
    >
      <animateTransform
        attributeName="transform"
        attributeType="xml"
        dur="0.6s"
        from="0 25 25"
        repeatCount="indefinite"
        to="360 25 25"
        type="rotate"
      />
    </path>
  </svg>
</div>
`;

exports[`<LoadingIndicator /> without any props renders correctly 1`] = `
<div
  className="LoadingIndicator LoadingIndicator-block"
>
  <svg
    className=""
    height="40px"
    version="1.1"
    viewBox="0 0 50 50"
    width="40px"
    x="0px"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    y="0px"
  >
    <path
      d="M25.251,6.461c-10.318,0-18.683,8.365-18.683,18.683h4.068c0-8.071,6.543-14.615,14.615-14.615V6.461z"
      fill="#FF6700"
    >
      <animateTransform
        attributeName="transform"
        attributeType="xml"
        dur="0.6s"
        from="0 25 25"
        repeatCount="indefinite"
        to="360 25 25"
        type="rotate"
      />
    </path>
  </svg>
</div>
`;
